apps/ejabberd/src/mongoose_cassandra_worker.erl

```diff
@@ -0,0 +1,494 @@
+%% Function: init(Args) -> {ok, State} |
+%%                         {ok, State, Timeout} |
+%%                         ignore               |
+%%                         {stop, Reason}
+%% Description: Initiates the server
+%%--------------------------------------------------------------------
+init([PoolName, Addr, Port, ClientOptions]) ->
+    async_spawn(Addr, Port, ClientOptions),
+    State = #state{pool_name=PoolName},
+    {ok, State}.
+
+init_connection(ConnPid, Conn, State=#state{pool_name=PoolName}) ->
+    erlang:monitor(process, ConnPid),
+    mongoose_cassandra_sup:register_worker(PoolName, self()),
+    put(query_refs_count, 0),
+    State#state{
+        conn=Conn,
+        query_refs=dict:new(),
+        query_refs_count=0,
+        prepared_queries=dict:new()}.
+
+%%--------------------------------------------------------------------
+%% Function: %% handle_call(Request, From, State) -> {reply, Reply, State} |
+%%                                      {reply, Reply, State, Timeout} |
+%%                                      {noreply, State} |
+%%                                      {noreply, State, Timeout} |
+%%                                      {stop, Reason, Reply, State} |
+%%                                      {stop, Reason, State}
+%% Description: Handling call messages
+%%--------------------------------------------------------------------
+handle_call(_, _From, State=#state{conn=undefined}) ->
+    {reply, no_connection, State};
+handle_call({cql_query, Module, QueryName, Params}, From,
+    State=#state{conn=Conn}) ->
+    case get_prepared_query(Conn, Module, QueryName, State) of
+        {ok, PreparedQuery, State2} ->
+            QueryRef = execute_prepared_query(Conn, PreparedQuery, Params),
+            {noreply, save_query_ref(From, QueryRef, State2)};
+        {error, Reason} ->
+            {reply, {error, Reason}, State}
+    end;
+handle_call({cql_batch, Module, BatchType, Queries}, From,
+    State=#state{conn=Conn}) ->
+    case new_batch_queries(Conn, Module, Queries, State) of
+        {ok, BatchQueries, State2} ->
+            QueryRef = run_batch(Conn, BatchType, BatchQueries),
+            {noreply, save_query_ref(From, QueryRef, State2)};
+        {error, Reason, State2} ->
+            {reply, {error, Reason}, State2}
+    end;
+handle_call(_, _From, State=#state{}) ->
+    {reply, ok, State}.
+
+
+%%--------------------------------------------------------------------
+%% Function: handle_cast(Msg, State) -> {noreply, State} |
+%%                                      {noreply, State, Timeout} |
+%%                                      {stop, Reason, State}
+%% Description: Handling cast messages
+%%--------------------------------------------------------------------
+
+handle_cast(_, State=#state{conn=undefined}) ->
+    {noreply, State};
+handle_cast({async_cql_query, Module, QueryName, Params},
+    State=#state{conn=Conn}) ->
```

Comment:
They are not written (yet?). But I think adding `space` around `=` or other operators makes things more readable. That's rather general advice and not Erlang-specific.